nautobot/extras/context_managers.py

```diff
@@ -152,3 +159,57 @@ def web_request_context(
         for object_change in ObjectChange.objects.filter(request_id=change_context.change_id).iterator():
             enqueue_job_hooks(object_change)
             enqueue_webhooks(object_change)
+
+
+@contextmanager
+def object_changelogs_bulk_operation(objs, user, action):
+    def get_objectchange_for_instance(instance, context_state, action):
+        objectchange = instance.to_objectchange(action)
+        objectchange.user = user
+        objectchange.request_id = context_state.change_id
+        objectchange.change_context = context_state.context
+        objectchange.change_context_detail = context_state.context_detail[:CHANGELOG_MAX_CHANGE_CONTEXT_DETAIL]
+        return objectchange
+
+    with transaction.atomic():
+        # Disable automatic ChangeLog creation on signal to enhance performance.
+        # Multiple object updates lead to excessive database calls.
+        # Instead, we'll manually manage a bulk ChangeLog creation post-update.
+        prev_state = change_context_state.set(None)
+
+        if action == ObjectChangeActionChoices.ACTION_DELETE:
```

Comment:
It "feels" like this function should, for completeness, handle ACTION_CREATE as well as ACTION_UPDATE and ACTION_DELETE. Maybe it'd be useful in some of the DCIM views where we bulk-create device components?